code/modules/mob/living/basic/lavaland/goldgrub/goldgrub_ai.dm

```diff
@@ -108,10 +108,15 @@
 	if(!dig_ability.IsAvailable())
 		return
 
-	var/mob/living/target = controller.blackboard[BB_BASIC_MOB_CURRENT_TARGET]
+	var/has_target = controller.blackboard_key_exists(BB_BASIC_MOB_CURRENT_TARGET)
 
 	//a storm is coming or someone is nearby, its time to escape
-	if(currently_underground || !currently_underground && storm_approaching || !QDELETED(target))
+	if(currently_underground)
```

Comment:
1 small thingy this would mean that during storms, goldgrubs would keep digging out and in. my intention was the grub will remain underground until the storm passes but if u want to change it then its fine

Reply:
Worry not it calls this a few lines above:
```
if(currently_underground && storm_approaching)
		return SUBTREE_RETURN_FINISH_PLANNING
```

It _does_ act pretty weird when burrowing to escape someone (on live right now) though, basically doing the popping up and down thing you mention here. I think it loses its target after burrowing or something, was going to make a future PR to investigate 😔

Reply:
my mistake i didnt see that